modules/cassandra/src/main/java/org/testcontainers/containers/CassandraContainer.java

```diff
@@ -151,28 +144,23 @@ public String getPassword() {
     }
 
     /**
-     * Get configured Cluster
+     * Get configured Session
      *
      * Can be used to obtain connections to Cassandra in the container
+     *
+     * @return A Configured {@link CqlSession}
      */
-    public Cluster getCluster() {
-        return getCluster(this, enableJmxReporting);
+    public CqlSession getSession() {
+        return getSession(this);
     }
 
-    public static Cluster getCluster(ContainerState containerState, boolean enableJmxReporting) {
-        final Cluster.Builder builder = Cluster.builder()
-            .addContactPoint(containerState.getHost())
-            .withPort(containerState.getMappedPort(CQL_PORT));
-        if (!enableJmxReporting) {
-            builder.withoutJMXReporting();
-        }
+    public static CqlSession getSession(ContainerState containerState) {
+        final CqlSessionBuilder builder = CqlSession.builder()
+            .addContactPoint(new InetSocketAddress(containerState.getHost(), containerState.getMappedPort(CQL_PORT)))
+            .withLocalDatacenter(DATACENTER);
         return builder.build();
     }
```

Comment:
Returning a third-party class from our convenience method is what's now given us this breaking change headache. It's a pattern we're trying to avoid with new modules for this exact reason.

Could we consider eliminating `com.datastax.*` types from the API of this class?

e.g.:
* delete the old `getCluster()` method as you've done
* do not add a `getSession` method
* add a new `getContactPoint()` method that simply returns an `InetSocketAddress`, and a `getDataCenter()` method that returns the `DATACENTER` constant

It would be slightly more work for users, e.g. they would have to do:
```
final CqlSessionBuilder builder = CqlSession.builder()
            .addContactPoint(myCassandraContainer.getContactPoint())
            .withLocalDatacenter(myCassandraContainer.getDatacenter());
        return builder.build();
```

but our public API from this class would just consist of JDK types, and we'd never have to make breaking changes again should (for example) the cassandra driver change classes/package names in v5.

If we're going to have to have a breaking change here, why not do it in a way that avoids future breaking changes?

BTW you'll notice I'm suggesting this be instance methods, not static methods - I am really quite surprised that these were ever static methods.

Reply:
@rnorth Thank you for the review and the comments!
I see your point about Testcontainers concerns and I think you are right. I'm a bit tied up with a few other things at the moment, but I will revisit this as soon as I can and try to implement the changes you suggest. I'll also try to update the Cassandra Testcontainer documentation if necessary.